OpenDreamRuntime/Objects/Types/DreamList.cs

```diff
@@ -1263,6 +1263,91 @@ public IEnumerable<DreamObjectTurf> GetTurfs() {
     }
 }
 
+// mob.contents, obj.contents list
+public sealed class MovableContentsList(DreamObjectDefinition listDef, DreamObjectMovable movable) : DreamList(listDef, 0) {
+    private readonly DreamObjectMovable _movable = movable;
+    public override DreamValue GetValue(DreamValue key) {
+        if (!key.TryGetValueAsInteger(out var index))
+            throw new Exception($"Invalid index into movable contents list: {key}");
+
+
+        if (index < 1 || index > _movable.ChildCount)
+            throw new Exception($"Out of bounds index on movable contents list: {index}");
+
+
+        using (var childEnumerator = _movable.ChildEnumerator) {
+            while (index >= 1) {
+                var current = childEnumerator.MoveNext(out EntityUid child);
+
+                if (index == 1) {
+                    if (AtomManager.TryGetMovableFromEntity(child, out var childObject))
+                        return new DreamValue(childObject);
+                    else
+                        throw new Exception($"Invalid child in movable contents list: {child}");
+                }
+
+                index--;
+            }
+        }
+
+        throw new Exception($"Out of bounds index on movable contents list after iterating: {key}");
+    }
+
+    public override List<DreamValue> GetValues() {
+        List<DreamValue> values = [];
```

Comment:
```suggestion
        List<DreamValue> values = new List<DreamValue>(_movable.ChildCount);
```
Pre-allocate the list size